scipy/signal/tests/test_signaltools.py

```diff
@@ -3793,24 +3794,28 @@ def test_spaced_1dperiod(self, xp):
 
         assert strength.ndim == 0
         assert phase.ndim == 0
-        assert_almost_equal(strength, targ_strength)
-        assert_almost_equal(phase, 2 * np.pi * targ_phase)
+        assert_almost_equal(xp.asarray(strength), xp.asarray(targ_strength))
+        rtol_kw = {'rtol': 5e-7} if is_torch(xp) else {}
+        xp_assert_close(xp.asarray(phase, dtype=xp.float64),
+                        xp.asarray(2 * xp.pi * targ_phase, dtype=xp.float64),
+                        **rtol_kw)
```

Comment:
probably better to turn the dtype check off rather than coercing the real result?

Reply:
<rant> I'm pretty sure I had a bunch of review comments asking to change `check_dtype=False` into an `xp.asarray`</rant>
Care to come up with a preferred pattern? To me personally, they are all equally ugly, so I'd be happy to roll with whatever works.

Reply:
- `xp.asarray(..., dtype=...)` should be used on the expected array when we know what dtype to expect.
- If, for some reason, it is too difficult to specify which dtype we expect, it is better to set `check_dtype=False` which acts as a TO-DO.
- Coercing the dtype of the actual result is worse because it is a much less obvious (and less searchable) TO-DO.

> I'm pretty sure I had a bunch of review comments

I don't think I'll have ever asked for the dtype of the actual result to be coerced, but please ignore that if I did.

Reply:
The issue here is taht both `phase` and `targ_phase` can be either arrays or python scalars.

Reply:
> both phase and targ_phase can be either arrays or python scalars.

As in, they can differ in this same test? Do we return scalars just for NumPy? If so, it should be possible to handle that with an `is_numpy` check.